Automattic/pocket-casts-android
# Do not allow to set podcast folder UUID to home folder UUID

## Description

Our app's logic relies on home folder UUID being `null` when sorting or doing other operations. However, there are way to update podcasts with the home folder UUID. I changed the behavior of `Podcast` entity to prevent this. Ideally, I'd prefer to make the property not nullable but there's too much internal logic assuming that `null == homeFolderUuid`. Trying to change it in this PR would probably introduce more bugs than not.

## Testing Instructions

1. Set podcast sorting order by `Episode release date`.
2. Disable `Sync settings` feature flag.
3. Subscribe to a podcast.
4. Check that the podcast is displayed in the `Podcasts` tab.
5. Refresh the profile.
6. Enable `Sync settings` feature flag.
7. Refresh the profile.
8. The subscribed podcast should be displayed in the `Podcasts` tab.

## Checklist
 
#### I have tested any UI changes...
